src/nodes/ImageView.vue

```diff
@@ -212,24 +252,30 @@ export default {
 	},
 	methods: {
 		async init() {
-			const [url, fallback] = this.$imageResolver.resolve(this.src)
-			return this.loadImage(url).catch((e) => {
-				if (fallback) {
-					return this.loadImage(fallback)
+			const candidates = this.$attachmentResolver.resolve(this.src)
+			return this.load(candidates)
+		},
+		async load(candidates) {
+			const candidate = candidates.shift()
```

Comment:
I liked, but we are mutating the `candidates` argument.
It can generate unexpected behaviors in the future.

```suggestion
		async load(candidates) {
		    candidates = [...candidates] // prevent argument mutation
			const candidate = candidates.shift()
```

What do you think?

Reply:
while we are at it...how about:
```js
const [candidate, ...rest] = candidates
```
So the meaning of candidates does not change within the function.

Or even

```js
...
  load(...candidates)
...
async load(candidate, ...fallbacks) {
  ...
  load(...fallbacks)
}
```

Reply:
@vinicius73 :sweat: Nice catch.
@max-nextcloud I prefer your first one which is slightly more explicit.
Done.